### DISCUSSION TITLE
Newby Basic Help

### DISCUSSION THREAD

**Discussion by xxBXSD** [canopen-python/canopen#349] [Q&A]
Hi Folks, I am trying to work my way through the examples here and gain an understanding but I am getting stuck at the very beginning. For example, what are we actually trying to do with the simple_ds402_node.py file? Are we creating a virtual network and then simulating a device on that network or do we have a network with a real device and the EDS file is providing the information for that device? I know that this may be a silly question but there is no basic description of the overall concept.
When I look inside the file I can understand connecting to the Network etc. and adding a node (real/simulated?), then the description says that we are going to read some data but you immediately WRITE values to the node??????
# Read a variable using SDO
node.sdo[0x1006].raw = 1
node.sdo[0x100c].raw = 100
node.sdo[0x100d].raw = 3

Are you setting values in the node so that you can read them? (if so, you should probably explain what you are dong!)

Again, I apologize if I have no idea but I am a very experienced control systems engineer and I still cant make any sense here!

Regards Simon

**Comment by acolomb** [comment]
I guess that's just a typo in the example script comment.  Of course it's *setting* values, not reading them.

The example is written assuming you have a real device on a real CAN network to talk to, described by an EDS file.
